src/cargo/core/compiler/context/compilation_files.rs

```diff
@@ -434,9 +434,7 @@ fn compute_metadata<'a, 'cfg>(
     unit.target.name().hash(&mut hasher);
     unit.target.kind().hash(&mut hasher);
 
-    if let Ok(rustc) = cx.config.rustc() {
-        rustc.verbose_version.hash(&mut hasher);
-    }
+    cx.build_config.rustc.verbose_version.hash(&mut hasher);
```

Comment:
In my perception, it seemed that `build_config` should probably stay immutable over the lifetime of the `Context` (and in fact that is enforced, I think, by #5358).

Reply:
Hm, not sure I understand this comment :) `build_config` is not mutated here at all?